### ISSUE TITLE
Effect settings vertical scroll bar missing

### ISSUE THREAD

**Issue by scratchg0lfr** [smeighan/xLights#3777]
NOTE: IF YOU DO NOT ATTACH A SEQUENCE THAT DEMONSTRATES YOUR PROBLEM THEN THERE IS A HIGH PROBABILITY YOUR ISSUE WILL JUST BE CLOSED AND NOT INVESTIGATED.

**Describe the bug**
When the effecct settings window is sized so that an effects "settings" do not all fit into the window, a vertical scroll bar allows the user to scroll up and down within the sized window.  The vertical scroll bar is missing when a sequence is opened with certain perspectives that make the effects window to small to show the settings of the effect.  The user can workaround the issue by resizing the effects window.  The issue seems to be per effect, as a resize on "warp" does not fix the issue with "shockwave".

**To Reproduce**
Attached and rgbeffects file with a perspective in it called bug so the window layout can be reproduced.
[xlights_bug.zip](https://github.com/smeighan/xLights/files/11165394/xlights_bug.zip)

1. Place the attached xlights_rgbeffects.xml and xlights_networks.xml (null controller) in a directory
2. Open xlights
3. Temporarily change the show directory to the folder where the files in step 1 are located
4. Click view>perspecitives and choose "bug"
5. Click to the sequence tab
6. Open the attached scrollbug.xsq sequence
7. The vertical scoll bar is missing on the effects window (see screenshot 1)
8. Resize the window by adjusting the size of all windows on the left
9. Scroll bar appears for the effect (see screenshot 2)
10. Choose another effect sequence
11. Veritcal scroll bar disappears
12. Resize the window by adjusting the size of all windows on the left
13. Scroll bar appears for the effect
14. Choose an effect that has already had it's window resized and the scroll bar appears.

**Expected behavior**
The vertical scroll bar should appear without resizing the window

**Screenshots**

![bug1](https://user-images.githubusercontent.com/67355195/230247073-6816a5b4-1d1c-49a0-b6a6-3d86d35e4858.png)
![bug2](https://user-images.githubusercontent.com/67355195/230247088-ef680fbd-4d9a-4812-ad5f-8def4e701c5a.png)

**Versions (please complete the following information):**
 - OS: Windows 11
 - xLights version 2023.4 and 2023.5.2 nightly-741d825

**Additional context**

**Attachments**
Zip file contains a sequence to demo the bug and an rgbeffects file to show the window layout perspective in which the bug appears.

**Comment by merryoncherry** [comment]
In my ongoing work to tweak the Ripple effect, I noticed this also.
Am investigating what the difference is between the effect panels, but would welcome any tips from experienced devs.
An additional observation is that some panels creep wider over time, wonder if it is related to the same problem that affects the buttons in the middle of the submodels dialog.

**Comment by AzGilrock** [comment]
I hope you have better luck than I did trying to fix it.   Found that issue many years ago and tried several things to fix it.   I had it happen on the DMX effect.   Seemed like it fixed itself when you do an action that requires the panel to update.   I tried to force initial updates and redraws and nothing fixed it.   I think I asked for help and other devs didn’t have luck either..

**Comment by scratchg0lfr** [comment]
I'm completely new to xlights code.  Just poking around and comparing ColoursPanel.cpp lines 335-366 to EffectsPanel.cpp lines 67-85.  I don't experience the issue with the color panel, so figured I'd compare it.  I probably have no business trying to fix it but you dont learn anything without trying.

**Comment by scratchg0lfr** [comment]
Added the following to line 78 of EffectsPanel.cpp
sw->SetScrollRate(5, 5);

Seems to be fixed.
![image](https://user-images.githubusercontent.com/67355195/230308623-7f12e4bc-9211-4051-9ad0-bd7a29f9e598.png)

**Comment by merryoncherry** [comment]
One of the wonderful things about life is that we can have more than one problem.

I couldn't repro your original problem with shockwave so I can't tell if that change helped.  (It didn't hurt anything, though.)

My problem with ripple was caused by an explicit size for that panel specifically.  Not sure how that got there, but something I did in codeblocks seems to have inadvertently introduced it.
